packages/store/addon/-private/system/adapter-cache.ts

```diff
@@ -0,0 +1,74 @@
+import normalizeModelName from './normalize-model-name';
+import { assert} from '@ember/debug';
+import { getOwner } from '@ember/application';
+import { isPresent } from '@ember/utils';
+import { set } from '@ember/object';
+
+export class AdapterCache {
+    _adapterCache: { [s: string]: any };
+    _store: any;
+
+    constructor(store: any) {
+        this._adapterCache = Object.create(null);
+        this._store = store;
+    }
+
+    adapterFor(modelName: string): any {
```

Comment:
following new `Adapter` type above
```suggestion
    adapterFor(modelName: string): Adapter {
```